### ISSUE TITLE
Mouse X/Y movements are inconsistent on start (rare)

### ISSUE THREAD

**Issue by tunbridgep** [libratbag/libratbag#911]
**Information**
- ratbagd version: 0.12
- Piper version: 0.3
- Distribution: Ubuntu MATE 19.10 Eoan
- Kernel: Linux 5.3.0-26-generic x86_64 GNU/Linux
- Mouse model: Logitech G600

**Describe the bug**
Sometimes, when starting my PC, my mouse will have different sensitivities in each direction - that is, it will move in the X and Y directions at different speeds. So moving it slightly to the left will move much further than moving the same distance upwards.

Changing profiles and then changing back using piper is enough to resolve the bug.

**To Reproduce**
This bug seems to happen completely at random when starting my PC, so this repro isn't exactly the most reliable
1. Start/Restart PC
2. Move mouse.
3. If it's working normally, go to step 1
4. Notice it moving oddly
5. Change profile, then change back. I do this with piper but it should be doable with the cli as well
6. Mouse movement is now fixed

**Logs**
This is probably useless, as I just tried 15 times in a row and was unable to reproduce the error, but here's what my log says on start

````raw
Initializing libratbag
ratbag debug: New device: Plantronics Plantronics GameCom 780
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 047f:c010
ratbag debug: New device: USB Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 04d9:0169
ratbag debug: New device: USB Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 04d9:0169
ratbag debug: New device: Logitech Gaming Mouse G600
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver logitech_g600
ratbag debug: hidraw info: bus 0x03 vendor 0x46d product 0xffffc24a
ratbag debug: Logitech Gaming Mouse G600 is device '/dev/hidraw2'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x46d product 0xffffc24a
ratbag debug: Logitech Gaming Mouse G600 is device '/dev/hidraw3'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: - HID report ID 01
ratbag debug: - HID report ID 80
ratbag debug: - HID report ID f6
ratbag debug: - HID report ID f0
ratbag debug: - HID report ID f1
ratbag debug: - HID report ID f2
ratbag debug: - HID report ID f3
ratbag debug: - HID report ID f4
ratbag debug: - HID report ID f5
ratbag debug: - HID report ID f6
ratbag debug: - HID report ID f7
ratbag raw: feature get:   f3 ad 7f a8 00 04 00 00 00 00 00 00 00 02 08 18 28 40 00 00 00 00 00 00 02 00 00 00 00 00 00 01 00 00 02 00 00 03 00 00 04 00 00 05 00 00 17 00 00 00 00 00 00 00 00 00 00 59 00 00 5a 00 00 5b 00 00 5c 00 00 5d 00 00 5e 00 00 5f 00 00 60 00 00 61 00 00 54 00 00 55 00 00 56 00 ff ff 01 00 00 02 00 00 03 00 00 04 00 00 05 00 00 17 00 00 13 00 00 14 00 00 00 01 1e 00 01 1f 00 01 20 00 01 21 00 01 22 00 01 23 00 01 24 00 01 25 00 01 26 00 01 27 00 01 2d 00 01 2e
ratbag debug: Unknown data in profile 0
ratbag debug:   profile->unknown1:   00 00 00 00 00
ratbag debug:   profile->unknown2:   00 00 00 00 00 00 02 00 00 00 00 00 00
ratbag raw: feature get:   f4 a4 00 00 00 00 00 00 00 00 00 01 08 02 08 18 28 40 00 00 00 00 00 00 02 00 00 00 00 00 00 01 00 00 02 00 00 03 00 00 04 00 00 05 00 00 17 00 00 00 02 05 14 00 00 00 00 00 00 00 5a 00 00 5b 00 00 5c 00 00 5d 00 00 5e 00 00 5f 00 00 60 00 00 61 00 00 62 00 00 56 00 00 57 ff ff ff 01 00 00 02 00 00 03 00 00 04 00 00 05 00 00 17 00 00 00 02 05 14 00 00 00 01 59 00 01 5a 00 01 5b 00 01 5c 00 01 5d 00 01 5e 00 01 5f 00 01 60 00 01 61 00 01 62 00 01 56 00 01 57
ratbag debug: Unknown data in profile 1
ratbag debug:   profile->unknown1:   00 00 00 00 00
ratbag debug:   profile->unknown2:   00 00 00 00 00 00 02 00 00 00 00 00 00
ratbag raw: feature get:   f5 00 ff 00 01 04 00 00 00 00 00 01 08 02 08 18 28 40 00 00 00 00 00 00 02 00 00 00 00 00 00 01 00 00 02 00 00 03 00 00 04 00 00 05 00 00 15 00 00 13 00 00 14 00 00 00 00 00 00 00 1f 00 00 20 00 00 21 00 00 22 00 00 23 11 00 00 00 00 00 05 00 00 12 00 00 00 00 08 04 00 00 ff ff ff 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
ratbag debug: Unknown data in profile 2
ratbag debug:   profile->unknown1:   00 00 00 00 00
ratbag debug:   profile->unknown2:   00 00 00 00 00 00 02 00 00 00 00 00 00
ratbag raw: feature get:   f0 0a 00 00
ratbag debug: driver match found: Logitech G600
hidraw2: "Logitech Gaming Mouse G600", 3 profiles
ratbag debug: New device: Logitech Gaming Mouse G600
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver logitech_g600
ratbag debug: hidraw info: bus 0x03 vendor 0x46d product 0xffffc24a
ratbag debug: Logitech Gaming Mouse G600 is device '/dev/hidraw2'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
DBus server ready
````

**Additional context**
I have added piper to my "Startup Applications" but I doubt this is the cause of the issue

I apologise for this issue being so vague in places (such as with the logs). It is really difficult if not impossible to reproduce on demand, and as such, I probably can't be as helpful as I would like.

**Comment by stephanlachnit** [comment]
Maybe you have a profile with different X/Y resolutions, which for some reason will be loaded if restarted. Can reproduce this when you re-plug the device as well? And can you send a list of all profiles/resolutions stored on your device?

**Comment by tunbridgep** [comment]
Okay so, I have gotten more information on this. I have done a LOT of experimentation with my PC over the last month, and I have noticed the following:

1. This issue happens on boot in ANY version of Linux, including versions that do not have libratbag installed. On Arch Linux I had the same problem
2. I have even had this problem in my Windows 10 dual-boot, which I set up this month specifically to test this. This is both with and without the (pretty crappy) software suite for my mouse installed
3. I have even had this issue in my BIOS setup screen once

This leads me to conclude that there is actually something wrong with the Logitech G600 itself, either a bug in it's actual firmware, or an issue with mine specifically. Either way, this is not a libratbag issue, so this can be closed now.
